frontend/components/Editor.js

```diff
@@ -489,8 +493,8 @@ export class Editor extends Component {
                 if (cell_ids.length > 0) {
                     await update_notebook((notebook) => {
                         for (let cell_id of cell_ids) {
-                            if (this.state.cell_inputs_local[cell_id]) {
-                                notebook.cell_inputs[cell_id].code = this.state.cell_inputs_local[cell_id].code
+                            if (this.state.notebook.cell_inputs[cell_id]?.local_code) {
```

Comment:
Now this is also checking the notebook instead of some separate state, it's better to use the notebook parameter to check against (instead of `this.state` which could possibly have an older/different version of the state once synchronisation becomes more complex)